src/main/remote-sync/RemoteSyncManager.ts

```diff
@@ -294,17 +308,28 @@ export class RemoteSyncManager {
       );
     }
 
-    Logger.info(
-      `Received ${isArray(response.data) && response.data.length} fetched files`
-    );
+    if (Array.isArray(response.data)) {
+      Logger.info(`Received ${response.data.length} fetched files`);
+    } else {
+      Logger.info(
+        `Expected to receive an array of files, but instead received ${JSON.stringify(
+          response,
+          null,
+          2
+        )}`
+      );
+
+      throw new Error('Did not receive an array of files');
+    }
+
     const hasMore =
-      isArray(response.data) &&
+      Array.isArray(response.data) &&
```

Comment:
This check is already made a few lines above. If it needed to have `response.data ` typed the the check can be converted into a guard clause

Reply:
fixed